src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -452,4 +457,54 @@ private boolean verifyAuthenticator(final AccountManager am) {
 
         return false;
     }
+
+    /**
+     * Waits on AccountManager results, so it should not be called on main
+     * thread.
+     * 
+     * @throws IOException
+     * @throws AuthenticatorException
+     * @throws OperationCanceledException
+     */
+    @Override
+    public UserInfo[] getBrokerUsers() throws OperationCanceledException, AuthenticatorException,
+            IOException {
+
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            throw new IllegalArgumentException("Calling getBrokerUsers on main thread");
+        }
+
+        Account[] accountList = mAcctManager
+                .getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
+        Bundle bundle = new Bundle();
+        bundle.putBoolean(DATA_USER_INFO, true);
+
+        if (accountList != null) {
+
+            // get info for each user
+            UserInfo[] users = new UserInfo[accountList.length];
+            for (int i = 0; i < accountList.length; i++) {
+                AccountManagerFuture<Bundle> result = mAcctManager.updateCredentials(
+                        accountList[i], AuthenticationConstants.Broker.AUTHTOKEN_TYPE, bundle,
+                        null, null, null);
```

Comment:
The name of this method is a bit strange. Are you updating credentials as part of a request to get broker users?